cmd/lakectl/cmd/fs_upload.go

```diff
@@ -32,13 +32,16 @@ var fsUploadCmd = &cobra.Command{
 
 		ctx, stop := signal.NotifyContext(ctx, os.Interrupt, os.Kill)
 		defer stop()
-
-		stat, err := os.Stat(source)
-		if err != nil {
-			Die("failed to stat source", 1)
+		ignoreRecessive := true // Ignore recursive if source is the standard input or a file (not a directory)
+		if source != StdinFileName {
+			stat, err := os.Stat(source)
+			if err != nil {
+				Die("failed to stat source", 1)
+			}
+			ignoreRecessive = !stat.IsDir()
 		}
 
```

Comment:
Do we need all this?  The changes seem functionally identical to
```go
		if source == StdinFileName {
			recursive = false
		}
```
and then checking for a directory.

Reply:
But in order to check for a directory I need to stat the file and I can't stat if source != StdinFileName 
So I will need this `if` again.

Reply:
Not sure where we need the stat id `!recursive`.

Reply:
@arielshaqed PTAL